frame/society/src/lib.rs

```diff
@@ -1277,11 +1329,11 @@ impl<T: Config<I>, I: Instance> Module<T, I> {
 	/// Can fail when `MaxMember` limit is reached, but has no side-effects.
 	fn add_member(who: &T::AccountId) -> DispatchResult {
 		let mut members = <Members<T, I>>::get();
-		ensure!(members.len() < MaxMembers::<I>::get() as usize, Error::<T, I>::MaxMembers);
+		ensure!(members.len() < <MaxMembers<T, I>>::get() as usize, Error::<T, I>::MaxMembers);
 		match members.binary_search(who) {
 			// Add the new member
 			Err(i) => {
-				members.insert(i, who.clone());
+				members.try_insert(i, who.clone()).expect("members is less than MaxMembers; qed");
```

Comment:
I think we can remove an expect here by simply returning an error on the `try_insert` here, right?

Reply:
That's probably not the right behaviour, we know for sure that this operation MUST succeed, because of the pop that we did a couple of lines above, which will make the length less than the maximum.

Reply:
Oh, I get what you mean, we can even make the ensure disappear altogether and just use ?. Let me see if it's possible...